pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -347,21 +347,22 @@ func getPhase2Binding(vmi *v1.VirtualMachineInstance, iface *v1.Interface, netwo
 		return &SlirpBindMechanism{vmi: vmi, iface: iface, domain: domain}, nil
 	}
 	if iface.Macvtap != nil {
+		macvtapBindMech := &MacvtapBindMechanism{
+			vmi:              vmi,
+			iface:            iface,
+			domain:           domain,
+			podInterfaceName: podInterfaceName,
+		}
+
 		mac, err := retrieveMacAddress(iface)
```

Comment:
If we move this check before the "macvtapBindMech" declaration we will be able to do the "return &MacvtapBindMechanism{..."

Reply:
Done.